include/llvm/ADT/PointerUnion.h

```diff
@@ -152,7 +152,14 @@ namespace llvm {
       assert(is<PT1>() && "Val is not the first pointer");
       assert(get<PT1>() == Val.getPointer() &&
          "Can't get the address because PointerLikeTypeTraits changes the ptr");
+#ifdef __GNUC__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wcast-qual"
+#endif
       return (PT1 *)Val.getAddrOfPointer();
```

Comment:
I really really dislike using these push pop macros. Looking over at the current state of the LLVM project: https://github.com/llvm/llvm-project/blob/master/llvm/include/llvm/ADT/PointerUnion.h

It looks like this has become ` return const_cast<PT1*>( reinterpret_cast<const PT1*>(this->Val.getAddrOfPointer()));`

Maybe that will work?

Reply:
Yes, the const_cast route is preferable.